ITMO_FS/hybrid/IWSSr_SFLA.py

```diff
@@ -0,0 +1,342 @@
+
+        Parameters
+        ----------
+        estimator : object
+            A supervised learning estimator with a fit method
+        measure_iwssr : string or callable
+            A standard estimator metric (e.g. 'f1' or 'roc_auc') or a callable object / function with signature 
+            measure(estimator, X, y) which should return only a single value that would be used in the IWSSr algorithm.
+        measure_frogs : string or callable
+            A standard estimator metric (e.g. 'f1' or 'roc_auc') or a callable object / function with signature 
+            measure(estimator, X, y) which should return only a single value that would be used to measure the frog's
+            fitness value.
+        cv : int
+            Number of folds in cross-validation.
+        relief_iterations : int
+            Amount of iterations to do in the Relief algorithm.
+        sfla_m : int
+            Amount of memplexes in the SFLA algorithm.
+        sfla_n : int
+            Amount of frogs in each memplex in the SFLA algorithm.
+        sfla_q : int
+            Amount of frogs in each submemplex in the SFLA algorithm. Should be lower than sfla_n.
+        s : int
+            Maximum amount of features that can change during one leap in the SFLA algorithm.
+        iterations : int
+            Total iteration number in the SFLA algorithm.
+        iterations_leaps : int
+            Total amount of leaps that each memplex would try to do during each iteration.
+        seed : int
+            Python random seed.
+
+
+        See Also
+        --------
+        For more details see `this paper <https://www.nature.com/articles/s41598-019-54987-1/>`_.
+
+        Examples
+        --------
+        >>> from ITMO_FS.hybrid import IWSSr_SFLA
+        >>> from sklearn.datasets import make_classification
+        >>> from sklearn.preprocessing import KBinsDiscretizer
+        >>> from sklearn.linear_model import LogisticRegression
+        >>> algo = IWSSr_SFLA(LogisticRegression())
+        >>> X, y = make_classification(n_informative=5, n_redundant=8, n_classes=3)
+        >>> est = KBinsDiscretizer(n_bins=10, encode='ordinal', strategy='uniform')
+        >>> est.fit(X)
+        >>> X = est.transform(X)
+        >>> algo.fit(X, y)
+        >>> print(algo.selected_features_)
+    """
+    def __init__(self, estimator, measure_iwssr='accuracy', measure_frogs='accuracy', cv=3, relief_iterations=None, sfla_m=5, 
+        sfla_n=20, sfla_q=8, s=3, iterations=20, iterations_leaps=10, seed=42):
+        self.estimator = estimator
+        self.measure_iwssr = measure_iwssr
+        self.measure_frogs = measure_frogs
+        self.cv = cv
+        self.relief_iterations = relief_iterations
+        self.sfla_m = sfla_m
+        self.sfla_n = sfla_n
+        self.sfla_q = sfla_q
+        self.s = s
+        self.iterations = iterations
+        self.iterations_leaps = iterations_leaps
+        self.seed = seed
+
+    def generate_frogs(self, weights):
```

Comment:
I think these functions should be shadowed

Reply:
true, fixed